osf/external/spam/tasks.py

```diff
@@ -0,0 +1,25 @@
+import re
+from framework.celery_tasks import app as celery_app
+from website.settings import DOMAIN_REGEX
+from django.contrib.contenttypes.models import ContentType
+
+
+@celery_app.task()
+def check_resource_for_domains(guid, content):
```

Comment:
Ack, this is based on an outdated comment from your first PR pass. If you and @adlius prefer the way you had it before (with more logic in the SpamMixin), I'm find with that. You'd largely reproduced this, just with the spam-check separated out into its own function instead of in the main loop.